src/avro.erl

```diff
@@ -130,6 +133,7 @@
              | [{name(), avro:out()}].
 
 -type codec_options() :: [proplists:property()].
+-type decoder_options() :: map().
```

Comment:
there is no need to add another type, please update `codec_options()` instead

Reply:
Could you clarify this comment a little more? Do you suggest to use proplist instead of map for decoder options or you meant adding something like ` | map()` in the dialzyer spec.

Reply:
ok. now I realise that it's perhaps not a good idea to have encoder/decoder options mixed.
Please update `codec_options()` to `encoder_options() :: prolists:proplist()`
And keep `decoder_options()` as it is, however, public APIs should still allow proplist as input,
so we need to either update API specs with `proplists:proplist() | decoder_options()`
or make another union type for it.

Please also be more specific about map fields.
i.e. write this type spec as:
```
-type decoder_options() :: #{ encoding => avro_binary | avro_json,
                            , map_type => proplist | map
                            , ....
                            }
```

Reply:
maybe export `-type codec_options() :: encoder_options() | decoder_options().` to be backward compatible
but we do not use it in our specs anymore.

Reply:
request for changes @ananthakumaran

Reply:
I have moved the definition to avro_internal.hrl and did some changes so we there won't be any need to expose the structure to the user. The user will still supply the option in the old code_options() type and that will be cast to maps and passed to other modules.